classes/external.php

```diff
@@ -298,4 +299,175 @@ public static function create_course_returns() {
         );
     }
 
+    /**
+     * Enrolls and unenrolls a user into a formations course.
+     *
+     * @param int $courseid The ID of the course to enroll the user into.
+     * @param int $userid The ID of the user to enroll.
+     * @param string $action The action to perform. Either 'enroll' or 'unenroll'.
+     * @param int $roleid The role id to assign to the user.
+     *
+     * @return array The result of the operation.
+     */
+    public static function edit_enrollment($courseid, $userid, $action = '', $roleid = 5) {
+        global $DB;
+
+        // Initialize the variables to validate the parameters.
+        $pluginname = 'enrol_coursepilot';
+        $validactions = ['enroll', 'unenroll'];
+        $validroles = [3, 4, 5];
```

Comment:
Are there constants for these numbers?

Reply:
Are not defined constants that can be used as Moodle doesn't provide predefined constants for role IDs like ROLE_ID_STUDENT, anyways I can use `get_archetype_roles` functions to retrieve roles by their archetypes